### PR TITLE
Bumps msys2 to 20201109

### PR DESCRIPTION
**Pull request by mjvankampen**
Specify library name and version:  **msys2/20201109**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/msys2/all/conanfile.py [hunk 1]
```diff
@@ -24,6 +24,8 @@ class MSYS2Conan(ConanFile):
     def configure(self):
         if self.settings.os_build != "Windows":
             raise ConanInvalidConfiguration("Only Windows supported")
+        if self.settings.arch_build == "x86" and self.version >= version("20201109"):
```
Comment (madebr) [inline recipes/msys2/all/conanfile.py:27]: Are we ready to use `arch/os` instead of `arch/os`?

File: recipes/msys2/all/conandata.yml [hunk 2]
```diff
@@ -32,3 +32,11 @@ sources:
              "https://sourceforge.net/projects/msys2/files/Base/x86_64/msys2-base-x86_64-20200517.tar.xz",
       ]
       sha256: "c4443113497acb2d2e285d40b929fc55f33f8f669902595ecdf66a655b63dc60"
+  "20201109":
+    - url: ""
+      sha256: ""
```
Comment (Croydon) [inline recipes/msys2/all/conandata.yml:37]: Why did you add two empty entries?

File: recipes/msys2/all/conanfile.py (hunk 1 above)
Comment (mjvankampen) [reply recipes/msys2/all/conanfile.py:27]: will try

File: recipes/msys2/all/conandata.yml (hunk 2 above)
Comment (mjvankampen) [reply recipes/msys2/all/conandata.yml:37]: To keep the data model consistent and not have to handle that in the recipe. The two empty lines represent the x86 files which are not available for this version.

**Comment by mjvankampen** [comment]
Closed in favour of #4286
